src/ChangeStream/ResourceStack.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/*
+ * This file is part of the puli/repository package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Puli\Repository\ChangeStream;
+
+use Puli\Repository\Api\Resource\PuliResource;
+use RuntimeException;
+use Webmozart\Assert\Assert;
+
+/**
+ * Represents a versionned stack of resources for a given path.
+ *
+ * You can access different versions of a resource using a ChangeStream
+ * that will return you resource stacks.
+ *
+ * @since  1.0
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class ResourceStack
+{
+    /**
+     * @var array
+     */
+    private $stack;
+
+    /**
+     * @param array $stack
+     */
+    public function __construct($stack)
+    {
+        Assert::isArray($stack, 'Built resource stack must be an array.');
+
+        $this->stack = $stack;
+    }
+
+    /**
+     * Get the last version from the stack.
+     *
+     * @return PuliResource
+     */
+    public function getCurrentVersion()
```

Comment:
Right now, the stack's wording is a bit ambiguous. `getCurrentVersion()` returns a resource, `getVersion()` too but `getAvailableVersions()` returns integers.

I think this class would be more useful if we do:

* `getCurrentVersion()` returns the latest version number
* `getVersions()` returns all version numbers
* `getCurrent()` returns the resource for the current version
* `getFirst()` returns the resource for the first version
* `get($version)` returns the resource for a specific version
* `toArray()` returns all resources indexed by version

Then "version" consistently refers to the integer number that corresponds to a resource at a specific point in time.